src/Tec/Plugin.php

```diff
@@ -100,26 +137,901 @@ public function register() {
 
 		// Do the settings.
 		// TODO: Remove if not using settings
-		$this->get_settings();
+		//$this->get_settings();
 
 		// Start binds.
 
+		add_filter( 'tec_tickets_commerce_attendee_post_type_args', [ $this, 'tc_attendees_label' ] );
+		add_filter( 'tec_tickets_commerce_order_post_type_args', [ $this, 'tc_orders_label' ] );
+
+		add_filter( 'tec_events_custom_tables_v1_tracked_meta_keys', [ $this, 'modify_tracked_meta_keys' ] );
 
+		add_filter( 'wp_all_import_is_post_to_create', [ $this, 'maybe_create_post' ], 10, 3 );
+
+		add_action( 'pmxi_update_post_meta', [ $this, 'maybe_skip_post_meta' ], 10, 3 );
+
+		add_action( 'pmxi_saved_post', [ $this, 'maybe_update_post' ], 10, 3 );
 
 		// End binds.
 
 		$this->container->register( Hooks::class );
 		$this->container->register( Assets::class );
 	}
 
+	/**
+	 * Check whether a post should to be imported or not.
+	 * We only do this for post types that must have a connection.
+	 * - Check for data validity.
+	 * - Check if the connection exists.
+	 *
+	 * @param bool  $continue_import True to import, false to skip import.
+	 * @param array $data            Array of data to import.
+	 * @param int   $import_id       The ID of the import.
+	 *
+	 * @return bool
+	 */
+	public function maybe_create_post( bool $continue_import, array $data, int $import_id ): bool {
+		if (
+			$data['posttype'] == 'tribe_rsvp_tickets'
```

Comment:
Use `in_array` here. Easier to maintain and read.
```php
if ( in_array( $data['posttype'], $post_types, true) ) { //... }
```

Reply:
Here you're accessing the `posttype` key on the `$data` array assuming it exists: never assume a key will exist in an array. Here and elsewhere, the first check should be an `isset( $array['key'] )` one. Bail if the key does not exist.

Reply:
To improve readability, use an early return pattern in the method body.
It's not always possible, but when it is, as is the case here, it can improve readablity.

```php
public function maybe_create_post( bool $continue_import, array $data, int $import_id ): bool {
	$supported_post_types = [
		'tribe_rsvp_tickets',
		'tribe_rsvp_attendees',
		// ...
	];

	if ( ! ( isset( $data['posttype'] ) && in_array( $data['posttype'], $supported_post_types, true ) ) ) {
		return true;
	}

	$this->add_to_log( "<strong>THE EVENTS CALENDAR EXTENSION: WPAI ADD-ON:</strong>" );

	if ( ! $this->check_data_validity( $data ) ) {
		return false;
	}

	if ( ! apply_filters( 'tec_labs_wpai_force_import_' . $data['posttype'], false, $data, $import_id ) ) {
		return false;
	}

	// ...

	return $this->check_relation_exists( $data );
}
```
The "pyramid of death of nested conditions makes debugging the code more difficult and increased the complexity.

Reply:
Addressed these to the best of my knowledge.